raiden-ts/src/path/epics.ts

```diff
@@ -1,48 +1,154 @@
 /* eslint-disable @typescript-eslint/camelcase */
-import { Observable, of, combineLatest, from, EMPTY, merge } from 'rxjs';
+import { combineLatest, defer, EMPTY, from, merge, Observable, of } from 'rxjs';
 import {
-  filter,
-  mergeMap,
-  publishReplay,
-  first,
   catchError,
   concatMap,
-  map,
-  withLatestFrom,
-  timeout,
   debounceTime,
-  pluck,
+  delay,
   distinctUntilChanged,
+  filter,
+  first,
   groupBy,
-  switchMap,
+  map,
+  mergeMap,
+  pluck,
+  publishReplay,
   scan,
   startWith,
+  switchMap,
   tap,
-  delay,
+  timeout,
+  withLatestFrom,
 } from 'rxjs/operators';
 import { fromFetch } from 'rxjs/fetch';
-import { isActionOf, ActionType } from 'typesafe-actions';
-import { bigNumberify, BigNumber } from 'ethers/utils';
-import { Zero, Two } from 'ethers/constants';
+import { ActionType, isActionOf } from 'typesafe-actions';
+import { BigNumber, bigNumberify, hexDataLength, hexlify, toUtf8Bytes } from 'ethers/utils';
+import { HashZero, Two, Zero } from 'ethers/constants';
 import { Event } from 'ethers/contract';
-import { isNil } from 'lodash';
+import { get, isNil } from 'lodash';
 
 import { RaidenAction } from '../actions';
 import { RaidenState } from '../state';
 import { RaidenEpicDeps } from '../types';
 import { getPresences$ } from '../transport/utils';
 import { messageGlobalSend } from '../messages/actions';
-import { PFSCapacityUpdate, MessageType } from '../messages/types';
-import { signMessage } from '../messages/utils';
+import { MessageType, PFSCapacityUpdate } from '../messages/types';
+import { MessageTypeId, signMessage } from '../messages/utils';
 import { channelDeposited } from '../channels/actions';
 import { ChannelState } from '../channels/state';
 import { channelAmounts } from '../channels/utils';
-import { Address, UInt, Int, decode } from '../utils/types';
-import { losslessStringify, losslessParse } from '../utils/data';
+import { Address, decode, Int, Signature, UInt } from '../utils/types';
+import { encode, losslessParse, losslessStringify } from '../utils/data';
 import { getEventsStream } from '../utils/ethers';
-import { pathFind, pathFound, pathFindFailed, pfsListUpdated } from './actions';
+import {
+  pathFind,
+  pathFindFailed,
+  pathFound,
+  pfsListUpdated,
+  udcBalanceFetch,
+  udcBalanceFetchFailed,
+  udcBalanceUpdate,
+} from './actions';
 import { channelCanRoute, pfsInfo, pfsListInfo } from './utils';
-import { PathResults, Paths } from './types';
+import { IOU, LastIOUResults, PathResults, Paths, PFS } from './types';
+import { concat } from 'ethers/utils/bytes';
+import { Signer } from 'ethers';
+
+const makeIOU = (sender: Address, receiver: Address, chainId: UInt<32>, oneToNAddress: Address) =>
+  ({
+    sender: sender,
+    receiver: receiver,
+    chain_id: chainId,
+    amount: Zero as UInt<32>,
+    one_to_n_address: oneToNAddress,
+    expiration_block: Zero as UInt<32>,
+    signature: HashZero as Signature,
+  } as IOU);
```

Comment:
As I commented on the `Signed` approach, this is not good. `HashZero` isn't a valid signature. Also,  the type assertion here can trick you sometimes. Instead, prefer putting `IOU` or `Signed<IOU>` in the return type of the function, and then just return the object. TypeScript will then make sure your object matches the return type or complain earlier otherwise.

Reply:
Thanks, I will change accordingly :)

Reply:
Also, as much as possible when creating objects, to create them ready. You can use this factory function by passing all needed parameters or create it in place (should be created in a single place, I suppose, right?), but try to make it ready. Zero isn't valid amount or expiration_block, and the thing with signature I already stated.